modules/navier_stokes/test/tests/finite_volume/cns/mms/1d/tests

```diff
@@ -0,0 +1,13 @@
+[Tests]
+  issues = '#16758'
+  design = 'CNSFVHLLCBase.md'
+  [1d-hllc]
+    type = PythonUnitTest
+    input = test.py
+    test_case = Test1DHLLC
+    requirement = 'The system shall be able to solve compressible fluid flow kernels for mass, momentum, and energy with the addition of diffusion and display first order convergence for all variables when using a HLLC scheme for the convection terms.'
```

Comment:
so 2 questions:
- what happens when you just slap a diffusion kernel onto HLLC? Because the scheme is meant for Euler equations, it doesnt look at diffusion of the shockwave
- it seems the no-diffusion-hllc-mms.i is not used by any test? Or am I missing it

Reply:
> what happens when you just slap a diffusion kernel onto HLLC? Because the scheme is meant for Euler equations, it doesnt look at diffusion of the shockwave

When running a real problem in which you actually have a shockwave, I myself do not know. I do know:

- there is quite a bit of literature about attempting to incorporate viscosity/conduction into characteristic/Riemann-style treatments of the Navier-Stokes equations. I also know that
- `rhoCentralFoam` does operator splitting for momentum advection/viscosity and energy advection/conduction. They do this because of time-step limitations imposed by the diffusion/conduction term for explicit schemes. They, however, do not do any special spatial discretization of viscosity/conduction nor do they do any modification of the Kurganov-Tadmor advective fluxes
- The approximation order as shown by this MMS test does not appear to be affected by the presence of `FVDiffusion`.


>     * it seems the no-diffusion-hllc-mms.i is not used by any test? Or am I missing it

You are correct. I will remove it. The `mms/1d-with-bcs` directory contains HLLC MMC tests with no `FVDiffusion`

Reply:
MMS results with diffusion are encouraging. 
From Toro, it seems either operator splitting or some sort of upwinding of the source terms are what is being considered for dealing with source terms. Since we do neither of those things, I m a little worried that our approach will be an issue for someone doing a real problem, and they wont be too happy when we tell them after they've modeled it. 

How do you think we can document that we do not have a special treatment of diffusion or source terms and that it could be necessary even though we survived without so far ?

Reply:
> From Toro, it seems either operator splitting or some sort of upwinding of the source terms are what is being considered for dealing with source terms.

Upwinding is a characteristic based concept. If you don't use a characteristic-based discretization of advection, e.g. if you use a centered scheme like Kurganov-Tadmor, then I think it makes total sense to have a centered discretization of source terms. As to how to discretize sources when using a characteristic-based approach to advection ... I honestly think by nature you're trying to fit a round peg in a square hole. To me you just shouldn't do a characteristic based approach with sources because the sources alter the equations. I've seen attempts to incorporate the sources into the eigendecomposition but I was never left feeling super-convinced. THM uses centered discretizations of sources in conjunction with HLLC. Maybe @joshuahansel or @andrsd has thoughts on it.

Reply:
In THM, we do not have a good solution for the source term discretization so far. We have not looked into the alternatives of the central source discretization. As far as the concern about users using it without knowing the consequences, I'd recommend you just have the code issue a warning when source terms are detected with an upwind discretization.

Reply:
Yeah I was mentioning this for HLLC not for KT. I m perfectly happy with the treatment of sources for KT. 
Thanks for the insight!